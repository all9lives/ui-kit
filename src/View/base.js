import { css } from 'styled-components'
import Unit from '../utils/Unit'
import Spacing from '../utils/Spacing'

const alignItemsMapping = {
  column_left: 'align-items: flex-start;',
  column_center: 'align-items: center;',
  column_right: 'align-items: flex-end;',
  row_top: 'align-items: flex-start;',
  row_middle: 'align-items: center;',
  row_bottom: 'align-items: flex-end;'
}

const justifyContentMapping = {
  column_top: 'justify-content: flex-start;',
  column_middle: 'justify-content: center;',
  column_bottom: 'justify-content: flex-end;',
  row_left: 'justify-content: flex-start;',
  row_center: 'justify-content: center;',
  row_right: 'justify-content: flex-end;'
}

const getAlignStyles = ({ direction, align }) => {
  const alignList = align.split(' ')
  return `${alignItemsMapping[`${direction}_${alignList[0]}`] || alignItemsMapping[`${direction}_${alignList[1]}`] || ''}
${justifyContentMapping[`${direction}_${alignList[0]}`] || justifyContentMapping[`${direction}_${alignList[1]}`] || ''}`
}

const base = css`
  flex-direction: ${props => props.direction};
  ${props => getAlignStyles(props)}
  margin: ${props => Unit.parse(props.margin, (margin) => Spacing[margin] || margin)};
  padding: ${props => Unit.parse(props.padding, (padding) => Spacing[padding] || padding)};
`

export default base
