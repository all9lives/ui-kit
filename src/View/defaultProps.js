export default {
  align: '',
  margin: 0,
  padding: 0
}
