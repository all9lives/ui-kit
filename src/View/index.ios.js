import View from './index.android'

export default View
