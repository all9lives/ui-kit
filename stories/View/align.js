import React from 'react'
import createStory from '../../storybook/createStory'
import View from '../../src/View'
import Text from '../../src/Text'

const squareStyle = { width: 200, height: 200, backgroundColor: 'white' }

createStory('View.align', module)
  .add('with align `top left`', () => (
    <View align='top left' style={squareStyle}>
      <Text>Top Left</Text>
    </View>
  ))
  .add('with align `top center`', () => (
    <View align='top center' style={squareStyle}>
      <Text>Top Center</Text>
    </View>
  ))
  .add('with align `top right`', () => (
    <View align='top right' style={squareStyle}>
      <Text>Top Right</Text>
    </View>
  ))
  .add('with align `middle left`', () => (
    <View align='middle left' style={squareStyle}>
      <Text>Middle Left</Text>
    </View>
  ))
  .add('with align `middle center`', () => (
    <View align='middle center' style={squareStyle}>
      <Text>Middle Center</Text>
    </View>
  ))
  .add('with align `middle right`', () => (
    <View align='middle right' style={squareStyle}>
      <Text>Middle Right</Text>
    </View>
  ))
  .add('with align `bottom left`', () => (
    <View align='bottom left' style={squareStyle}>
      <Text>Bottom Left</Text>
    </View>
  ))
  .add('with align `bottom center`', () => (
    <View align='bottom center' style={squareStyle}>
      <Text>Bottom Center</Text>
    </View>
  ))
  .add('with align `bottom right`', () => (
    <View align='bottom right' style={squareStyle}>
      <Text>Bottom Right</Text>
    </View>
  ))
