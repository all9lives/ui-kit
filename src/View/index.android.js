import styled from 'styled-components/native'
import { View as NativeView } from 'react-native'
import base from './base'
import defaultProps from './defaultProps'
import { filterProps } from '../utils/Misc'

export const Base = filterProps(NativeView, ['direction', 'margin', 'padding'])
Base.displayName = 'View'

const View = styled(Base)`
  ${base}
`
View.defaultProps = {
  ...defaultProps,
  direction: 'column'
}

export default View
