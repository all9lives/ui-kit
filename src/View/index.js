import styled from 'styled-components'
import Unit from '../utils/Unit'
import base from './base'
import defaultProps from './defaultProps'

const View = styled.div`
  ${base}
  display: flex;
  height: ${props => Unit.convert(props.height)};
  width: ${props => props.fullWidth ? '100%' : Unit.convert(props.width)};
`
View.defaultProps = {
  ...defaultProps,
  width: 'auto',
  height: 'auto',
  direction: 'row'
}

export default View
