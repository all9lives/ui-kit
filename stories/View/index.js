import React from 'react'
import createStory from '../../storybook/createStory'
import View from '../../src/View'
import Text from '../../src/Text'
import './align'

createStory('View', module)
  .add('with direction `column`', () => (
    <View direction='column'>
      <Text>Row 1</Text>
      <Text>Row 2</Text>
    </View>
  ))
  .add('with direction `row`', () => (
    <View direction='row'>
      <Text>Column 1</Text>
      <Text>Column 2</Text>
    </View>
  ))
  .add('with margin `10`', () => (
    <View margin={10}>
      <Text>Margin 10</Text>
    </View>
  ))
  .add('with padding `10`', () => (
    <View padding={10}>
      <Text>Padding 10</Text>
    </View>
  ))
  .add('with width `120`', () => (
    <View width={120}>
      <Text>Width 120</Text>
    </View>
  ))
  .add('with height `120`', () => (
    <View height={120}>
      <Text>Height 120</Text>
    </View>
  ))
  .add('with grow `1:2`', () => (
    <View direction='row'>
      <View grow={1}><Text>Grow 1</Text></View>
      <View grow={2}><Text>Grow 2</Text></View>
    </View>
  ))
