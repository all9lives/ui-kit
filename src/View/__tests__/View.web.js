/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import View from '../index.js'

it('render snapshot for `View` on `web`', () => {
  const wrapper = shallow(
    <View>Apple</View>
  )
  expect(wrapper).toMatchSnapshot()
})
