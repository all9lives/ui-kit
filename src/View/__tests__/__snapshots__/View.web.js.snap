// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View.web renders a snapshot with \`align:top,left\` 1`] = `
.c0 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin: 0px;
  padding: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: autopx;
  width: autopx;
}

<div
  className="c0"
  direction="row"
  height="auto"
  width="auto"
/>
`;

exports[`View.web renders a snapshot with \`direction:row\` 1`] = `
.c0 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 0px;
  padding: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: autopx;
  width: autopx;
}

<div
  className="c0"
  direction="row"
  height="auto"
  width="auto"
/>
`;

exports[`View.web renders a snapshot with \`fullWidth\` 1`] = `
.c0 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 0px;
  padding: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: autopx;
  width: 100%;
}

<div
  className="c0"
  direction="row"
  height="auto"
  width="auto"
/>
`;

exports[`View.web renders a snapshot with \`grow:1\` 1`] = `
.c0 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  margin: 0px;
  padding: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: autopx;
  width: autopx;
}

<div
  className="c0"
  direction="row"
  height="auto"
  width="auto"
/>
`;

exports[`View.web renders a snapshot with \`height:120\` 1`] = `
.c0 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 0px;
  padding: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 120px;
  width: autopx;
}

<div
  className="c0"
  direction="row"
  height={120}
  width="auto"
/>
`;

exports[`View.web renders a snapshot with \`margin:0.5\` 1`] = `
.c0 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 5px;
  padding: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: autopx;
  width: autopx;
}

<div
  className="c0"
  direction="row"
  height="auto"
  width="auto"
/>
`;

exports[`View.web renders a snapshot with \`padding:0.5\` 1`] = `
.c0 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 0px;
  padding: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: autopx;
  width: autopx;
}

<div
  className="c0"
  direction="row"
  height="auto"
  width="auto"
/>
`;

exports[`View.web renders a snapshot with \`width:120\` 1`] = `
.c0 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 0px;
  padding: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: autopx;
  width: 120px;
}

<div
  className="c0"
  direction="row"
  height="auto"
  width={120}
/>
`;
